### PR TITLE
AuthorizationRequest nonce

### PR DESCRIPTION
**Pull request by jhetrifork**
Added optional authorizationRequestNonce parameter to performOpenIDConnectLogin making it possible to define a custom nonce for the authorization request.

### CODE REVIEW

File: Sources/TIM/Protocols/TIMAuth.swift [hunk 1]
```diff
@@ -36,7 +36,7 @@ public protocol TIMAuth {
     ///   - presentingViewController: The view controller which the safari view controller should be presented on.
     ///   - completion: Invoked with access token after successful login (or with error)
     @available(iOS, deprecated: 13)
-    func performOpenIDConnectLogin(presentingViewController: UIViewController, completion: @escaping AccessTokenCallback)
```
Comment (kimdv) [inline Sources/TIM/Protocols/TIMAuth.swift:39]: Could you add an extension to `TIMAuth` with `func performOpenIDConnectLogin(presentingViewController: UIViewController, completion: @escaping AccessTokenCallback)`


Just so we don't break the API

Comment (jhetrifork) [reply Sources/TIM/Protocols/TIMAuth.swift:39]: I thought that if the param was optional, it would not break the api?

Comment (jhetrifork) [reply Sources/TIM/Protocols/TIMAuth.swift:39]: Ahh I get what you mean now. It is not possible to define a default value in the protocol. :).

File: Sources/TIM/AppAuth/AppAuthController.swift [hunk 2]
```diff
@@ -243,5 +238,34 @@ public final class AppAuthController: OpenIDConnectController {
...
+                redirectURL: self.credentials.redirectUri,
+                responseType: OIDResponseTypeCode,
+                additionalParameters: [:]
+            )
+        } else {
+            return OIDAuthorizationRequest(
+                configuration: config,
+                clientId: self.credentials.clientId,
+                clientSecret: nil,
+                scope: self.credentials.scopes.joined(separator: " "),
```
Comment (kimdv) [inline Sources/TIM/AppAuth/AppAuthController.swift:258]: We should use `OIDScopeUtilities.scopes(with: self.credentials.scopes)` here. 

https://github.com/openid/AppAuth-iOS/blob/98ed5e15ceedcacc0b3fd6c3521a22411acc2212/Source/AppAuthCore/OIDAuthorizationRequest.m#L178

File: Sources/TIM/AppAuth/AppAuthController.swift [hunk 3]
```diff
@@ -243,5 +238,34 @@ public final class AppAuthController: OpenIDConnectController {
             completion(result)
         }
     }
+    
+    /// Create AuthorizationRequest using the authorizationRequestNonce in case it is provided
+    private func createAuthorizationRequest(config: OIDServiceConfiguration, authorizationRequestNonce: String?) -> OIDAuthorizationRequest {
+        if authorizationRequestNonce == nil {
```
Comment (kimdv) [inline Sources/TIM/AppAuth/AppAuthController.swift:244]: I would do `if let authorizationRequestNonce = authorizationRequestNonce { ... } else { ... }` as it's more swift to do

File: Sources/TIM/Protocols/TIMAuth.swift (hunk 1 above)
Comment (jhetrifork) [reply Sources/TIM/Protocols/TIMAuth.swift:39]: I have added an extension function that defines a default parameter for authorizationRequestNonce, but not on without the authorizationRequestNonce param, should we have two separate extension functions?

Comment (kimdv) [reply Sources/TIM/Protocols/TIMAuth.swift:39]: It should actually be without `authorizationRequestNonce: String? = nil`

So

```swift
/// Wrapper of `performOpenIDConnectLogin(presentingViewController:completion:authorizationRequestNonce)` adding nil as default value for authorizationRequestNonce
     func performOpenIDConnectLogin(presentingViewController: UIViewController, completion: @escaping AccessTokenCallback) {
         self.performOpenIDConnectLogin(presentingViewController: presentingViewController, completion: completion, authorizationRequestNonce: nil)
     }
```

Comment (jhetrifork) [reply Sources/TIM/Protocols/TIMAuth.swift:39]: Ok. Thanks.

**Review by kimdv — APPROVED** [review]

**Pull request merged**